File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.java
class Solution {
    public int countKConstraintSubstrings(String s, int k) {
        int l=0, r=0, zeros=0, ones=0, res = 0;

        for(r=0; r < s.length(); r++){
            if(s.charAt(r) == '1') {
                ones +=1;
            }else{
                zeros +=1;
            }
            while(ones>k && zeros>k){
                if(s.charAt(l) == '1') {
                    ones -=1;
                }else{
                    zeros -=1;
                }
                l +=1;
            }
            res += r-l+1;
        }
        return res;
    }
}

File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.py
class Solution:
    def countKConstraintSubstrings(self, s: str, k: int) -> int:
        result = 0
        for i in range(0,len(s)):
            zero = 0
            ones = 0
            for j in range(i,len(s)):
                if s[j] == "0":
                    zero +=1
                else:
                    ones +=1
                if zero<= k or ones<=k:
                    result+=1
        return result